### PR TITLE
Updates MVS to better handle differing support in .NET Framework versions

### PR DESCRIPTION
**Pull request by jaffinito**
## Description

- Updates MVS to ignore any .NET Framework that we are not going to support.  Examples:  .NET 3.5, Silverlight, Xamarin
- Re-enables RestSharp
- Addresses some formatting issues

# Author Checklist
- [x] Unit tests, Integration tests, and Unbounded tests completed
- [ ] Performance testing completed with satisfactory results (if required)
- [ ] [Agent Changelog](/src/Agent/CHANGELOG.md) or [Lambda Agent changelog](/src/AwsLambda/CHANGELOG.md) updated 

# Reviewer Checklist
- [ ] Perform code review
- [ ] Pull request was adequately tested (new/existing tests, performance tests)
- [ ] Review Changelog

### CODE REVIEW

File: tests/Agent/MultiverseTesting/ReportBuilder/WikiMaker.cs [hunk 1]
```diff
@@ -2,196 +2,196 @@
...
-        private void WriteContent(string filePath, string content, bool shouldAppend)
-        {
-            using (var writer = new StreamWriter(filePath, append: shouldAppend))
-            {
-                writer.WriteLine(content);
-            }
-        }
-    }
-}
+//using System.Collections.Generic;
```
Comment (chynesNR) [inline tests/Agent/MultiverseTesting/ReportBuilder/WikiMaker.cs:5]: Is this file still needed?

File: tests/Agent/MultiverseTesting/ReportBuilder/DotNetFrameworkParser.cs [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+using System;
+using System.Collections.Generic;
+
+namespace ReportBuilder
+{
+    static public class DotNetMonikerNormalizer
+    {
+        private const string NetFramework = ".NET Framework 4.5.x to 4.8.x";
+        private const string NetStandard = ".NET Standard 2.0, 2.1";
+        private const string NetCore = ".NET Core 3.1 and .NET 5.0 to 7.0";
```
Comment (JcolemanNR) [inline tests/Agent/MultiverseTesting/ReportBuilder/DotNetFrameworkParser.cs:13]: Please add a comment indicating that this code is not currently used, and possibly comment the file out

File: tests/Agent/MultiverseTesting/ConsoleScanner/Program.cs [hunk 3]
```diff
@@ -75,6 +75,13 @@ public static void ProcessAssemblies(ScannerConfiguration configuration)
                 {
                     foreach (var intrumentedDllFileLocation in downloadedNugetInfo.InstrumentedDllFileLocations)
                     {
+                        // skip unusable frameworks like silverlight
+                        var targetFramework = Path.GetFileName(Path.GetDirectoryName(intrumentedDllFileLocation));
+                        if (!targetFramework.StartsWith("net") || targetFramework.StartsWith("net35"))
```
Comment (JcolemanNR) [inline tests/Agent/MultiverseTesting/ConsoleScanner/Program.cs:80]: Please review https://learn.microsoft.com/en-us/nuget/reference/target-frameworks#supported-frameworks and add ignores for any other undesirable target frameworks. 

Here are a few that I see:
- net11
- net20
- net40
- net403
- net4** up to < 4.6.2?
- netmf ? (.NET Microframework?)
- netcoreapp1.0
- netcoreapp1.1
- netcoreapp2.0
- netcoreapp2.1
- netcoreapp2.2
- netcoreapp3.0

File: tests/Agent/MultiverseTesting/ReportBuilder/styles.css [hunk 4]
```diff
@@ -12,12 +13,20 @@ tr:nth-child(even) {
     background-color: #e6e6e6;
 }
 
+tr:hover {
+    background-color: #D6EEEE;
```
Comment (JcolemanNR) [inline tests/Agent/MultiverseTesting/ReportBuilder/styles.css:17]: fancy! ❤️

File: tests/Agent/MultiverseTesting/ReportBuilder/HtmlMaker.cs [hunk 5]
```diff
@@ -252,5 +253,12 @@ private void WriteContent(string filePath, string content, bool shouldAppend)
                 writer.WriteLine(content);
             }
         }
+
+        private string MultilineMethodSignature(string methodSignature)
+        {
+            var parts = methodSignature.Split('(');
+            return $"<b>{parts[0]}</b>({parts[1].Replace(",", ", ")}";
```
Comment (JcolemanNR) [inline tests/Agent/MultiverseTesting/ReportBuilder/HtmlMaker.cs:260]: MUCH more readable 👍

**Review by JcolemanNR — CHANGES_REQUESTED** [review]
It looks like this will make it work for RestSharp! I think we should be more thorough on target frameworks that we support/don't support and have added a few comments about that.

Looking to the future, we will have to handle assemblies that have different instrumentation points for .NET Core and .NET Framework, but I don't think that needs to be solved immediately.

File: tests/Agent/MultiverseTesting/ConsoleScanner/Program.cs (hunk 3 above)
Comment (jaffinito) [reply tests/Agent/MultiverseTesting/ConsoleScanner/Program.cs:80]: Will do!

File: tests/Agent/MultiverseTesting/ReportBuilder/WikiMaker.cs (hunk 1 above)
Comment (jaffinito) [reply tests/Agent/MultiverseTesting/ReportBuilder/WikiMaker.cs]: Removed.

File: tests/Agent/MultiverseTesting/ReportBuilder/DotNetFrameworkParser.cs (hunk 2 above)
Comment (jaffinito) [reply tests/Agent/MultiverseTesting/ReportBuilder/DotNetFrameworkParser.cs]: Removed.

**Review by chynesNR — APPROVED** [review]

**Review by JcolemanNR — APPROVED** [review]
Looking good!
